core/services/ocr2/plugins/ocr2keeper/evm21/encoder.go

```diff
@@ -75,31 +76,47 @@ func (enc EVMAutomationEncoder21) Encode(results ...ocr2keepers.CheckResult) ([]
 		if !ok {
 			return nil, fmt.Errorf("failed to parse big int from upkeep id: %s", string(result.Payload.Upkeep.ID))
 		}
+
 		report.UpkeepIds[i] = id
 		report.GasLimits[i] = big.NewInt(0).SetUint64(result.GasAllocated)
 
 		triggerW := triggerWrapper{
 			BlockNum:  uint32(result.Payload.Trigger.BlockNumber),
 			BlockHash: common.HexToHash(result.Payload.Trigger.BlockHash),
 		}
+
 		switch getUpkeepType(id.Bytes()) {
 		case logTrigger:
-			trExt, ok := result.Payload.Trigger.Extension.(logprovider.LogTriggerExtension)
-			if !ok {
+			var trExt logprovider.LogTriggerExtension
+
+			switch typedExt := result.Payload.Trigger.Extension.(type) {
+			case logprovider.LogTriggerExtension:
+				trExt = typedExt
+			case string:
```

Comment:
Can we add a test case for string based extension?

Reply:
done